frontend/app/components/app-link.tsx

```diff
@@ -0,0 +1,23 @@
+import { Link, type LinkProps } from '@remix-run/react';
+import { useTranslation } from 'react-i18next';
+
+/**
+ * Props for the AppLink component.
+ */
+export type AppLinkProps = { locale?: 'en' | 'fr' } & LinkProps;
+
+/**
+ * A component that renders a localized link.
+ */
+export function AppLink({ locale, ...props }: AppLinkProps) {
```

Comment:
Are we all in agreement that the component file names will use kebab-case (ie. `app-link.tsx` instead of `AppLink.tsx`)?

Reply:
That's my preference, consistency is king. Why have a different convention for react component filenames vs non-component filenames?

Reply:
https://airbnb.io/javascript/react/#naming

Reply:
> Name your files using PascalCase, matching the component name. For example, if you have a component named UserCard, the file should be named UserCard.js.

https://www.linkedin.com/pulse/react-js-naming-convention-kristiyan-velkov/

Reply:
It's up the to the team to decide. I always been in the side of file name should match component name. 🤷

Reply:
https://remix.run/docs/en/main/file-conventions/routes

Reply:
I agree... for routes files use kekab... for Component use PascalCase or kebab-case?

Reply:
> https://remix.run/docs/en/main/file-conventions/routes

I'm good with these conventions - resolving

Reply:
`/app/components/app-link.tsx` or `/app/components/AppLink.tsx`

Reply:
For non-routes, Remix docs also use kebab-case.

Reply:
> `/app/components/app-link.tsx` or `/app/components/AppLink.tsx`

The former.